R/zchunk_L105.nh3_an_USA_S_T_Y.R

```diff
@@ -53,19 +55,63 @@ module_emissions_L105.nh3_an_USA_S_T_Y_DISABLED <- function(command, ...) {
+      select(-Source_Category_Raw, -Source_Category) %>%
+      gather(year, value, -sector, -fuel) %>%
+      mutate(year = as.numeric(year)) %>%
+      filter(year %in% NH3_HISTORICAL_YEARS) %>%
+      mutate(value = value * CONV_TST_TG) %>%  # convert from thousand short tons to Tg
+      dplyr::rename(emissions = value)
+
+    # Process FAO production data:
+    # Subset US data, aggregate to EPA sectors (all animal production)
+
+    L105.out_Mt_USA_an_C_Sys_Fd_Yh <- L107.an_Prod_Mt_R_C_Sys_Fd_Y %>%
+      filter(GCAM_region_ID == 1) %>%
+      group_by(year) %>%
+      summarize_at(vars(value), funs(sum ) ) %>%
+      dplyr::rename(production = value) %>%
+      mutate(year = as.numeric(year))
+
+    # Compute GHG emissions factors by dividing EPA inventory by FAO animal production
+    L105.nh3_tgmt_USA_an_Yh <-  L105.nh3_tg_USA_an_Sepa_F_Yh %>%
+      left_join_error_no_match(L105.out_Mt_USA_an_C_Sys_Fd_Yh, by = "year") %>%
+      mutate(value = emissions/production) %>%
+      select(-emissions, -production)
+
+    # Add historical years (use 1990 estimate for other historical years (1971 - 1989) )
+    extend_sector <- L105.nh3_tgmt_USA_an_Yh %>% filter(year == 1990) %>% select(sector) %>% as.character
+    extend_fuel <- L105.nh3_tgmt_USA_an_Yh %>% filter(year == 1990) %>% select(fuel) %>% as.character
+    extend_value <- L105.nh3_tgmt_USA_an_Yh %>% filter(year == 1990) %>% select(value)%>% as.numeric
+
+    L105.nh3_tgmt_USA_an_Yh <- L105.nh3_tgmt_USA_an_Yh %>%
+      bind_rows( tibble(sector = rep(extend_sector, times = length(NH3_EXTRA_YEARS)),
+                    fuel = rep(extend_fuel, times = length(NH3_EXTRA_YEARS)),
+                    year = NH3_EXTRA_YEARS,
+                    value = rep(extend_value, times = length(NH3_EXTRA_YEARS)) )  ) %>%
+      arrange(year)
+
     # Produce outputs
-    # Temporary code below sends back empty data frames marked "don't test"
-    # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+
+    L105.nh3_tgmt_USA_an_Yh %>%
+      add_title("NH3 emissions factors for animals by sector / technology / year estiamted from EPA NH3 emissions and FAO production statistics") %>%
+      add_units("Tg/Mt") %>%
+      add_comments("NH3 animal emission factors for all historical years are estimates from US values") %>%
+      add_comments("Historical years 1971 - 1989 are estimated as 1990 values") %>%
       add_legacy_name("L105.nh3_tgmt_USA_an_Yh") %>%
-      add_precursors("precursor1", "precursor2", "etc") %>%
-      # typical flags, but there are others--see `constants.R`
+      add_precursors("common/iso_GCAM_regID",
```

Comment:
I don't see this used anywhere above? If it's not, please remove here (and from inputs as well).